### ISSUE TITLE
Multiple color tweens

### ISSUE THREAD

**Issue by kujohn** [legomushroom/mojs#114]
Are there anyway to change multiple fill color in one tween such as:

```
fill: { 'red': 'pink', 'green': 'black', 'yellow': 'orange' }
```
It does not seem to work. I have to resort to mutiple `.then` callbacks to accomplish this. Am I doing something wrong?

**Comment by legomushroom** [comment]
Hey John!

Unfortunately, it is not possible and you need to rely on the `then` chains. The problem is that `javascript` does not give any reliable way to get the order of object keys, thus no order of animation will be preserved.

**Comment by kujohn** [comment]
I see. I have not looked at the source but I am guessing an array doesn't work as a structure?
If there are no major reason why, I could give it a shot and implement it using arrays.

**Comment by legomushroom** [comment]
@kujohn sure, could you tell me how do you see the `API` with arrays?

**Comment by kujohn** [comment]
Something like:
```
fill: [ '#eee', 'red', 'orange', 'black' ]
```

**Comment by legomushroom** [comment]
@kujohn looks nice 👍, what about `easing`? or maybe `delay`/`duration`?

**Comment by kujohn** [comment]
Yeah, I think if we have an order data type for these, we can introduce order matching for `easing/delay/duration` with the fill array. Or we can Have these metadata inside the object of the array. But I am not sure what the direction of mojs is heading. LMK.

**Comment by legomushroom** [comment]
My concern is that the ordered items will not be very readable. Maybe some kind of keyframes will do?

```
const html = new mojs.Html({
  fill: [
    { to: '#eee', easing: 'ease.out' },
    { to: 'red', easing: 'ease.in' },
    { to: 'orange', easing: 'sin.in' },
    { to: 'black', easing: 'quad.out' },
  ]
});
```
